query/src/sql/exec/mod.rs

```diff
@@ -199,4 +208,46 @@ impl PipelineBuilder {
 
         Ok(output_schema)
     }
+
+    fn build_aggregate(
+        &mut self,
+        aggregate: &AggregatePlan,
+        input_schema: DataSchemaRef,
+    ) -> Result<DataSchemaRef> {
+        let output_schema = input_schema.clone();
```

Comment:
Does this `output_schema` work? It looks not that reasonable to me.

Reply:
It seems to just need to be consistent with the input schema? Isn't it? 🤔

Such as: `select sum(a) from t group by a;` (table t only has a column `a`)
The schemas from bottom to up are the followings:

table_scan
```
    fields: [
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
    ],
```

aggregation plan
```
    fields: [
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
    ],
```

projection plan
```
    fields: [
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
        DataField {
            name: "\"?column?\"_1",
            data_type: Int64,
            nullable: true,
        },
    ],
```

Reply:
The aggregation functions should be added to `DataSchema` so it can be accessed by projection plan later.

Besides, aggregate operator would be splited into 2 stages: `AggregatePartial` and `AggregateFinal`, you may refer to: https://github.com/datafuselabs/databend/blob/ec0acd34ae8c058dcb59f60c5d596d3b2e48f29a/query/src/pipelines/new/pipeline_builder.rs#L101-L136

Reply:
> The aggregation functions should be added to `DataSchema` so it can be accessed by projection plan later.

The reason I didn't add the aggregation function to the data schema here before is because I see that `ProjectPlan` gets the aggregation function from its `Vec<ProjectItem>` when it processes schema:
https://github.com/datafuselabs/databend/blob/43af01f26652f745f8b4dbc256b8d21c0d37eabe/query/src/sql/exec/data_schema_builder.rs#L43

In the case: `select sum(a) from t group by a`, `ProjectPlan`'s output schema is the following though I didn't add aggregate functions to `AggregatePlan`'s output schema. `?column?` is the aggregate expr.
```
    fields: [
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
        DataField {
            name: "\"?column?\"_1",
            data_type: Int64,
            nullable: false,
        },
    ],
```

In fact, I think the right output schema for `ProjectPlan` should be the following:
```
    fields: [
        DataField {
            name: "sum(a)",
            data_type: Int64,
            nullable: false,
        },
    ],
```
Now I add aggregate functions to `DataSchema`, and the output schema for each operator is the following:

`PhysicalScan`
```
    fields: [
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
    ],
```

`Aggregate`
```
    fields: [
        DataField {
            name: "sum(\"a\"_0)",
            data_type: Int64,
            nullable: false,
        },
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
    ],
```

`Project`
```
    fields: [
        DataField {
            name: "sum(\"a\"_0)",
            data_type: Int64,
            nullable: false,
        },
        DataField {
            name: "\"a\"_0",
            data_type: Int32,
            nullable: false,
        },
        DataField {
            name: "\"?column?\"_1",
            data_type: Int64,
            nullable: true,
        },
    ],
```

I think `PhysicalScan` and `Aggregate` output schemas are right, but for `Project`, it should be 
```
    fields: [
        DataField {
            name: "sum(a)",
            data_type: Int64,
            nullable: false,
        },
    ],
```

Reply:
Oops, `align_data_schema` eliminates my worries